lib/Standalone/ToBlockLayoutAndBack.cpp

```diff
@@ -209,6 +167,102 @@ static LogicalResult BlockOpPreconditions(linalg::LinalgOp linalgOp) {
   return success();
 }
 
+static Value toPackLayoutRSCK_KCRSck(Location loc, Value input,
+                                     ArrayRef<OpFoldResult> tiles,
+                                     OpBuilder &builder) {
+  assert(tiles.size() == 2);
+  SmallVector<int64_t> innerDimsPos = {2, 3};
+  SmallVector<int64_t> outerDimsPerm = {3, 2, 0, 1};
+  return toPackLayoutImpl(loc, input, tiles, innerDimsPos, outerDimsPerm,
+                          builder);
+}
+
+// return true if the conv has stride != 1.
+template <typename CONVOP> static bool hasStride(CONVOP convOp) {
+  if (DenseIntElementsAttr strides = convOp.getStrides()) {
+    auto values = strides.getValues<APInt>();
+    if (llvm::any_of(values, [](const APInt &value) {
+          return value.getSExtValue() != 1;
+        })) {
+      return true;
+    }
+  }
+  return false;
+}
+
+//===----------------------------------------------------------------------===//
+// Conv2DNhwcHwcfOp
+//===----------------------------------------------------------------------===//
+// Original layout: [N][P][Q][K] += [N][H][W][C] * [R][S][C][K]
+// New      layout: [N][K'][P][Q][k] += [N][C'][H][W][c] * [K'][C'][R][S][c][k]
+FailureOr<linalg::GenericOp>
+mlir::linalgx::packConv2DNhwcHwcfOp(RewriterBase &rewriter,
+                                    linalg::Conv2DNhwcHwcfOp convOp,
+                                    ArrayRef<OpFoldResult> tiles) {
+  if (tiles.size() != 2)
+    return rewriter.notifyMatchFailure(convOp, "require 2 tile factors");
+
+  if (convOp.hasDynamicShape())
+    return rewriter.notifyMatchFailure(convOp, "require static shape");
+
+  if (convOp.hasBufferSemantics())
+    return rewriter.notifyMatchFailure(convOp, "require tensor semantics");
+
+  // TODO: (lorenzo) add the stride when you compose the affine map.
+  if (hasStride(convOp))
+    return rewriter.notifyMatchFailure(convOp, "require unit stride");
+
+  Location loc = convOp.getLoc();
+  MLIRContext *ctx = convOp.getContext();
+  SmallVector<Value, 2> reshapedInputTensors;
+
+  SmallVector<Value> inputOperands = convOp.getInputOperands();
+  SmallVector<Value> outputOperands = convOp.getOutputOperands();
+
+  // pack the image and the filter.
+  Value image = inputOperands[0];
+  Value packedImage = toPackLayoutNKPQk(loc, image, tiles[0], rewriter);
```

Comment:
These don't `replaceAllUses` because the shapes of the following linalgs that need the same tensor don't have the same shape, right?

This should be easy to do a pass later on to de-duplicate the calls.